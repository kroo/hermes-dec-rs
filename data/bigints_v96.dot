// CFG analysis for bigints_v96
// Generated from data/bigints_v96.hbc

digraph {
  rankdir=TB;
  node [shape=box, fontname="monospace"];

  subgraph cluster_function_0 {
    label = "Function 0";
    style = filled;
    color = lightgrey;

    f0_n0 [ label = "Block 0 (PC 0-11)\l  0: DeclareGlobalVar  \"a\"\l  1: GetGlobalObject   r0\l  2: TryGetById        r3, r0, 1, \"BigInt\"\l  3: LoadConstUndefined r2\l  4: LoadConstBigInt   r1, 123456789012345678901234567890\l  5: Call2             r1, r3, r2, r1\l  6: PutById           r0, r1, 1, \"a\"\l  7: TryGetById        r2, r0, 2, \"console\"\l  8: GetByIdShort      r1, r2, 3, \"log\"\l  9: GetByIdShort      r0, r0, 4, \"a\"\l  10: Call2             r0, r1, r2, r0\l" ]
    f0_n1 [ label = "Block 1 (PC 11-12)\l  0: Ret               r0\l" ]
    f0_n2 [ label = "EXIT" ]

    f0_n0 -> f0_n1
    f0_n1 -> f0_n2
  }

}
