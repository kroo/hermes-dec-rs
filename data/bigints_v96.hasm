Bytecode File Information:
  Bytecode version number: 96
  Source hash: dd710631b7985f55a4ea12394251a6c3f1dfc0ce
  Function count: 1
  String count: 5
  BigInt count: 1
  String Kind Entry count: 2
  RegExp count: 0
  Segment ID: 0
  CommonJS module count: 0
  CommonJS module count (static): 0
  Function source count: 0
  Bytecode options:
    staticBuiltins: 0
    cjsModulesStaticallyResolved: 0

Global String Table:
s0[ASCII, 2..7] #1C35E808: global
s1[ASCII, 0..2] #00018270: log
s2[ASCII, 6..6] #6FA5A548: a
s3[ASCII, 8..13] #629A2BFD: BigInt
s4[ASCII, 14..20] console

Global BigInt Table:
  0[0..12]: 123456789012345678901234567890

Array Buffer:

Function<global>0(1 params, 12 registers, 0 symbols):
  DeclareGlobalVar  "a"
  GetGlobalObject   r0
  TryGetById        r3, r0, 1, "BigInt"
  LoadConstUndefined r2
  LoadConstBigInt   r1, 123456789012345678901234567890
  Call2             r1, r3, r2, r1
  PutById           r0, r1, 1, "a"
  TryGetById        r2, r0, 2, "console"
  GetByIdShort      r1, r2, 3, "log"
  GetByIdShort      r0, r0, 4, "a"
  Call2             r0, r1, r2, r0
  Ret               r0

