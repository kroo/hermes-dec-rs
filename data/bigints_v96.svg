<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 13.1.0 (20250701.0955)
 -->
<!-- Pages: 1 -->
<svg width="526pt" height="430pt"
 viewBox="0.00 0.00 526.00 430.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 426)">
<polygon fill="white" stroke="none" points="-4,4 -4,-426 522,-426 522,4 -4,4"/>
<g id="clust1" class="cluster">
<title>cluster_function_0</title>
<polygon fill="lightgrey" stroke="lightgrey" points="8,-8 8,-414 510,-414 510,-8 8,-8"/>
<text xml:space="preserve" text-anchor="middle" x="259" y="-396.7" font-family="Times,serif" font-size="14.00">Function 0</text>
</g>
<!-- f0_n0 -->
<g id="node1" class="node">
<title>f0_n0</title>
<polygon fill="none" stroke="black" points="502.12,-381.5 15.88,-381.5 15.88,-166.5 502.12,-166.5 502.12,-381.5"/>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-364.2" font-family="monospace" font-size="14.00">Block 0 (PC 0&#45;11)</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-346.95" font-family="monospace" font-size="14.00"> &#160;0: DeclareGlobalVar &#160;&quot;a&quot;</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-329.7" font-family="monospace" font-size="14.00"> &#160;1: GetGlobalObject &#160;&#160;r0</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-312.45" font-family="monospace" font-size="14.00"> &#160;2: TryGetById &#160;&#160;&#160;&#160;&#160;&#160;&#160;r3, r0, 1, &quot;BigInt&quot;</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-295.2" font-family="monospace" font-size="14.00"> &#160;3: LoadConstUndefined r2</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-277.95" font-family="monospace" font-size="14.00"> &#160;4: LoadConstBigInt &#160;&#160;r1, 123456789012345678901234567890</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-260.7" font-family="monospace" font-size="14.00"> &#160;5: Call2 &#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;r1, r3, r2, r1</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-243.45" font-family="monospace" font-size="14.00"> &#160;6: PutById &#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;r0, r1, 1, &quot;a&quot;</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-226.2" font-family="monospace" font-size="14.00"> &#160;7: TryGetById &#160;&#160;&#160;&#160;&#160;&#160;&#160;r2, r0, 2, &quot;console&quot;</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-208.95" font-family="monospace" font-size="14.00"> &#160;8: GetByIdShort &#160;&#160;&#160;&#160;&#160;r1, r2, 3, &quot;log&quot;</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-191.7" font-family="monospace" font-size="14.00"> &#160;9: GetByIdShort &#160;&#160;&#160;&#160;&#160;r0, r0, 4, &quot;a&quot;</text>
<text xml:space="preserve" text-anchor="start" x="23.88" y="-174.45" font-family="monospace" font-size="14.00"> &#160;10: Call2 &#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;r0, r1, r2, r0</text>
</g>
<!-- f0_n1 -->
<g id="node2" class="node">
<title>f0_n1</title>
<polygon fill="none" stroke="black" points="370.12,-130.5 147.88,-130.5 147.88,-88 370.12,-88 370.12,-130.5"/>
<text xml:space="preserve" text-anchor="start" x="155.88" y="-113.2" font-family="monospace" font-size="14.00">Block 1 (PC 11&#45;12)</text>
<text xml:space="preserve" text-anchor="start" x="155.88" y="-95.95" font-family="monospace" font-size="14.00"> &#160;0: Ret &#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;r0</text>
</g>
<!-- f0_n0&#45;&gt;f0_n1 -->
<g id="edge1" class="edge">
<title>f0_n0&#45;&gt;f0_n1</title>
<path fill="none" stroke="black" d="M259,-166.26C259,-157.74 259,-149.63 259,-142.36"/>
<polygon fill="black" stroke="black" points="262.5,-142.51 259,-132.51 255.5,-142.51 262.5,-142.51"/>
</g>
<!-- f0_n2 -->
<g id="node3" class="node">
<title>f0_n2</title>
<polygon fill="none" stroke="black" points="286,-52 232,-52 232,-16 286,-16 286,-52"/>
<text xml:space="preserve" text-anchor="middle" x="259" y="-29.32" font-family="monospace" font-size="14.00">EXIT</text>
</g>
<!-- f0_n1&#45;&gt;f0_n2 -->
<g id="edge2" class="edge">
<title>f0_n1&#45;&gt;f0_n2</title>
<path fill="none" stroke="black" d="M259,-87.85C259,-80.34 259,-71.69 259,-63.59"/>
<polygon fill="black" stroke="black" points="262.5,-63.83 259,-53.83 255.5,-63.83 262.5,-63.83"/>
</g>
</g>
</svg>
