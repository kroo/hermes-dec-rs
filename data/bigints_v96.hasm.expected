Bytecode File Information:
  Bytecode version number: 96
  Source hash: dd710631b7985f55a4ea12394251a6c3f1dfc0ce
  Function count: 1
  String count: 5
  BigInt count: 1
  String Kind Entry count: 2
  RegExp count: 0
  Segment ID: 0
  CommonJS module count: 0
  CommonJS module count (static): 0
  Function source count: 0
  Bytecode options:
    staticBuiltins: 0
    cjsModulesStaticallyResolved: 0

Global String Table:
s0[ASCII, 2..7]: global
i1[ASCII, 0..2] #1C35E808: log
i2[ASCII, 6..6] #00018270: a
i3[ASCII, 8..13] #6FA5A548: BigInt
i4[ASCII, 14..20] #629A2BFD: console

Global BigInt Table:
 0[ 0[12..0]: 1234567890123456... (30 decimal digits)

Function<global>0(1 params, 12 registers, 0 symbols):
    DeclareGlobalVar  "a"
    GetGlobalObject   r0
    TryGetById        r3, r0, 1, "BigInt"
    LoadConstUndefined r2
    LoadConstBigInt   r1, 1234567890123456... (30 decimal digits)n
    Call2             r1, r3, r2, r1
    PutById           r0, r1, 1, "a"
    TryGetById        r2, r0, 2, "console"
    GetByIdShort      r1, r2, 3, "log"
    GetByIdShort      r0, r0, 4, "a"
    Call2             r0, r1, r2, r0
    Ret               r0

Debug filename table:
  0: hbc_testcases/bigints.js

Debug file table:
  source table offset 0x0000: filename id 0

Debug source table:
  0x0000  function idx 0, starts at line 1 col 1
    bc 7: line 1 col 9 scope offset 0x0000 env none
    bc 19: line 1 col 15 scope offset 0x0000 env none
    bc 24: line 1 col 7 scope offset 0x0000 env none
    bc 30: line 2 col 1 scope offset 0x0000 env none
    bc 36: line 2 col 12 scope offset 0x0000 env none
    bc 41: line 2 col 13 scope offset 0x0000 env none
    bc 46: line 2 col 12 scope offset 0x0000 env none
  0x0043  end of debug source table

Debug scope descriptor table:
  0x0000  lexical parent:   none, flags:    , variable count: 0
  0x0003  end of debug scope descriptor table

Textified callees table:
  0x0000  entries: 0
  0x0001  end of textified callees table

Debug string table:
  0x0000  end of debug string table

